cub/cub/agent/agent_histogram.cuh

```diff
@@ -285,22 +285,22 @@ struct AgentHistogram
   SampleT* d_native_samples;
 
   /// The number of output bins for each channel
-  int (&num_output_bins)[NUM_ACTIVE_CHANNELS];
+  int* num_output_bins;
```

Comment:
question: this somehow seems less informative. Are there any downsides to storing references to `cuda::std::array` instead?

Reply:
If I can change the constructor of `AgentHistogram`, I can do that! I chose the pointer, so the ctor of `AgentHistogram` is still callable with a native array and with `cuda_std_array.data()`. If `AgentHistogram` stores a `cuda::std::array<...>&`, I would need to change the ctor to `cuda::std::array<...>&` as well to avoid converting between `cuda::std::array<...>&` and native arrays.

Reply:
Reverted back to the original version of `AgentHistogram`.